src/analyses/constant_propagator.cpp

```diff
@@ -76,16 +70,33 @@ void constant_propagator_domaint::transform(
   ai_baset &ai,
   const namespacet &ns)
 {
-  #ifdef DEBUG
+#ifdef DEBUG
+  std::cout << "Transform from/to:\n";
   std::cout << from->location_number << " --> "
             << to->location_number << '\n';
-  #endif
+#endif
 
 #ifdef DEBUG
-  std::cout << "before:\n";
+  std::cout << "Before:\n";
   output(std::cout, ai, ns);
 #endif
 
+  const constant_propagator_ait *cp=
+    dynamic_cast<constant_propagator_ait *>(&ai);
+
+  bool have_dirty=false;
+  bool ignore_unresolved_calls=false;
+
+  if(cp!=nullptr)
```

Comment:
Ok! It may be worthwhile adding comments about such bits of knowledge that only the developer has.